app/ota_status.py

```diff
@@ -80,31 +80,45 @@ def _initial_read_ota_status(self):
         """
         status = ""
         logger.debug(f"ota status file: {self._status_file}")
-        try:
-            if os.path.exists(self._status_file):
-                with open(self._status_file, mode="r") as f:
-                    status = f.readline().replace("\n", "")
-                    logger.debug(f"line: {status}")
-            else:
-                logger.warning(f"No OTA status file: {self._status_file}")
-        except:
-            logger.exception("OTA status read error!")
+        if os.path.exists(self._status_file):
+            with open(self._status_file, mode="r") as f:
+                status = f.readline().replace("\n", "")
+                logger.debug(f"line: {status}")
+        else:
+            logger.warning(f"No OTA status file: {self._status_file}")
+            self._gen_ota_status_file(self._status_file)
```

Comment:
以下で十分だと思います。(indentは浅いほうがいいです)
```suggestion
      with open(self._status_file, mode="r") as f:
          status = f.readline().replace("\n", "")
          logger.debug(f"line: {status}")
      except Exception:
          logger.warning(f"No OTA status file: {self._status_file}")
          self._gen_ota_status_file(self._status_file)
```

Reply:
status  = self._gen_ota_status_file(self._status_file) 
として、_gen_ota_status_fileがota_status を返すようにすれば、
https://github.com/tier4/ota-client/pull/31/files#diff-a9d0ddbcf0e88953b4d102f15deb8703f22d1de5d6afb8204cdf30a83a779bbaR81
の
status = ""
が不要になると思います。

Reply:
@obi-t4  
with openでは、ファイルがない場合のexceptionは処理されないのでは？
exception処理するには、try:を入れないとうまく行かないと思います。

Reply:
あれ、すいません try: が抜けていました。try - cexept です。